src/timepicker/timepicker.js

```diff
@@ -503,6 +504,13 @@ angular.module('mgcrea.ngStrap.timepicker', ['mgcrea.ngStrap.helpers.dateParser'
         controller.$parsers.unshift(function(viewValue) {
           // console.warn('$parser("%s"): viewValue=%o', element.attr('ng-model'), viewValue);
           var date;
+
+          // If the model needs to be updated to matter what, these validation routines
+          // can be overridden by setting the attribute overrideValidation eq true
+          if (defaults.overrideValidation === 'true'){
```

Comment:
All this does is force 'whatever was typed' to go back into the view model. That way it can be validated outside of the timepicker control.  In case for instance, i use 24 hr time format, or want to support seconds.  I can use momentjs and custom validation to ensure that a proper datetime object is set. Works similar to this section, but allows custom abilities.